.github/actions/set-up-mac/action.yml

```diff
@@ -0,0 +1,39 @@
+name: "Set up rust on mac"
+description: "Install the required tools for Mac runners"
+inputs:
+  IMAGE:
+    description: "Rust docker image"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Install with Hombrew
+      shell: bash
+      run: brew install protobuf rustup openssl pkg-config zlib xz zstd llvm jq curl gcc make cmake
+    - name: Set version
+      shell: bash
+      run: |
+        VERSION=$(echo $IMAGE | sed -E 's/.*:bullseye-([^-]+)-.*/\1/')
+        echo $VERSION
+        echo "VERSION=$VERSION" >> $GITHUB_ENV
```

Comment:
Could you please also add `NIGHTLY_VERSION` variable and install nightly with the specific version?